server/src/controllers/group.controllers.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, RequestHandler, Response } from 'express';
+import { Group } from '../models/group.model';
+import { GroupModel } from '../models/group.model';
+
+const sendResponse = (
+  res: Response,
+  statusCode: number,
+  payload: GroupModel[] | GroupModel | string
+) => {
+  return res.status(statusCode).json(payload);
+};
+
+const createGroup: RequestHandler = async (req: Request, res: Response) => {
+  const { ...toCreate } = req.body;
+
+  // Check if any fields are present. MongoDB will throw an error
+  // if no fields are present or if a required field is missing
+  if (Object.keys(toCreate).length > 0) {
+    try {
+      const data = await Group.create(toCreate);
+
+      if (data) {
+        sendResponse(res, 200, data);
+      } else {
+        sendResponse(res, 404, 'Group not created');
+      }
+    } catch (error: any) {
+      sendResponse(res, 500, error.message);
```

Comment:
not sure if we need to handle the case whereby error doesn't have a message property
i.e. `error?.message`